runtime/lib/ttnn/operations/data_movement/permute.cpp

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "permute.h"
+
+#include "tt/runtime/detail/logger.h"
+#include "ttnn/common/constants.hpp"
+
+#include <vector>
+
+namespace tt::runtime::ttnn::operations::data_movement {
+void run(const ::tt::target::ttnn::PermuteOp *op, ProgramContext &context) {
+  ProgramTensorPool &tensorPool = context.getTensorPool();
+
+  const ::ttnn::Tensor &in = tensorPool.at(op->in()->global_id());
+  DEBUG_ASSERT(in.is_allocated());
+
+  std::vector<int64_t> permutation(op->permutation()->begin(),
+                                   op->permutation()->end());
+
+  // Has to called with the verbose invoke, since calling it with composite=true
```

Comment:
Do we need to create an issue on the Metal side regarding this?

Reply:
Yeah, I created an issue last week and it seems they have a fix that will land on the main soon https://github.com/tenstorrent/tt-metal/pull/15881. That's why I moved this PR to draft, to not cause an issue during uplift.